<a href="https://colab.research.google.com/github/BeBrave-BeHumble/Coding_test/blob/main/Codeup/CodeUp_basic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 정수가 입력 되었을 때 짝수면 출력해보자.

In [ ]:
aa = [ print(i) for i in list(map(int, input().split())) if i % 2 == 0 ]

1 2 4
2
4


# 정수가 입력 되었을 때 짝(even)/홀(odd)을 출력해보자.

In [ ]:
for i in list(map(int, input().split())) :
    print('even') if i % 2 == 0 else print('odd')

1 2 8
odd
even
even


# 6067 : [기초-조건/선택실행구조] 정수 1개 입력받아 분류하기

0이 아닌 정수 1개가 입력되었을 때, 음(-)/양(+)과 짝(even)/홀(odd)을 구분해 분류해보자.
음수이면서 짝수이면, A
음수이면서 홀수이면, B
양수이면서 짝수이면, C
양수이면서 홀수이면, D
를 출력한다.

In [ ]:
a = int(input())

if (a < 0 and a % 2 == 0) : print('A')
elif (a < 0 and a % 2 != 0) : print('B')
elif (a > 0 and a % 2 == 0) : print('C')
else : print('D')

-2147483648
A


# 6069 : [기초-조건/선택실행구조] 평가 입력받아 다르게 출력하기

평가 내용
평가 : 내용
A : best!!!
B : good!!
C : run!
D : slowly~
나머지 문자들 : what?

In [ ]:
평가 = str(input())

if 평가 == 'A' : print('best!!!')
elif 평가 == 'B' : print('good!!')
elif 평가 == 'C' : print('run!')
elif 평가 == 'D' : print('slowly~')
else : print('what?')

A
best!!!


In [ ]:
평가 = str(input())

평가_매핑 = {'A': 'best!!!', 'B': 'good!!', 'C': 'run!', 'D': 'slowly'}
print(평가_매핑.get(평가, 'what?'))

A
best!!!


# 6071 : [기초-반복실행구조] 0 입력될 때까지 무한 출력하기

임의의 정수가 줄을 바꿔 계속 입력된다.
-2147483648 ~ +2147483647, 단 개수는 알 수 없다.

0이 아니면 입력된 정수를 출력하고, 0이 입력되면 출력을 중단해보자.

In [ ]:
n = 1

while n != 0 :
    n = int(input())
    if n != 0 : print(n)

7
7
4
4
2
2
3
3
0


In [ ]:
n = int(input())

while n != 0 :
    print(n)
    n = int(input())

7
7
4
4
3
3
5
5
0


In [ ]:
aa = [n for n in iter(lambda: int(input()), 0)]
print(aa)

# 이렇게하면 출력값을 list에 담아버림 -> 오답

7
4
5
3
0
[7, 4, 5, 3]


# 6072 : [기초-반복실행구조] 정수 1개 입력받아 카운트다운 출력하기1

In [ ]:
n = int(input())

while n > 0 :
  print(n)
  n -= 1

10
10
9
8
7
6
5
4
3
2
1


# 6073 : [기초-반복실행구조] 정수 1개 입력받아 카운트다운 출력하기2

In [ ]:
n = int(input())

while n > 0 :
    print(n-1)
    n -= 1

5
4
3
2
1
0


# 6074 : [기초-반복실행구조] 문자 1개 입력받아 알파벳 출력하기

a부터 입력한 문자까지 순서대로 공백을 두고 한 줄로 출력한다.

In [ ]:
for i in range(ord('a'), ord(str(input()))+1):
    print(chr(i), end=' ')

z
a b c d e f g h i j k l m n o p q r s t u v w x y z 

In [ ]:
alphabet_string = ' '.join(chr(i) for i in range(ord('a'), ord(str(input()))+1))
print(alphabet_string)

# 이게 조금 더 빠름

z
a b c d e f g h i j k l m n o p q r s t u v w x y z


In [ ]:
print(*range(4))

0 1 2 3


In [ ]:
alphabet = [chr(i) for i in range(ord('a'), ord('z')+1)]
print(alphabet)

['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']


In [ ]:
chr(print(*range(ord('a'), ord('z')+1)))

97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122


# 6075 : [기초-반복실행구조] 정수 1개 입력받아 그 수까지 출력하기1

0부터 그 수까지 줄을 바꿔 한 개씩 출력한다.

In [ ]:
kk = '\n'.join( str(i) for i in range(int(input())+1) )
print(kk)

# 이게 더 빠름

3
0
1
2
3


In [ ]:
for i in range(int(input())+1):
    print(i, end = '\n')

3
0
1
2
3


# 6077 : [기초-종합] 짝수 합 구하기(설명)

In [ ]:
result = 0
for i in range(int(input())+1):
    if i % 2 == 0 :
        result += i
print(result)

5
6


In [ ]:
result = 0
kk = [ i for i in range(int(input())+1) if i % 2 == 0 ]
print(sum(kk))

# 더 빠름

5
6


# 6078 : [기초-종합] 원하는 문자가 입력될 때까지 반복 출력하기

영문 소문자 'q'가 입력될 때까지
입력한 문자를 계속 출력하는 프로그램을 작성해보자.

In [ ]:
while True:
    x = str(input())
    print(x)
    if x == 'q' :
        break

d
d
s
s
q
q


In [ ]:
for x in iter(input, 'q'):
    print(x)

ㄴ
ㄴ
a
a
d
d
q


# 6079 : [기초-종합] 언제까지 더해야 할까?(py)

1, 2, 3 ... 을 계속 더해 나갈 때,
그 합이 입력한 정수(0 ~ 1000)보다 같거나 작을 때까지만
계속 더하는 프로그램을 작성해보자.

즉, 1부터 n까지 정수를 계속 더해 나간다고 할 때,
어디까지 더해야 입력한 수보다 같거나 커지는 지를 알아보고자하는 문제이다.


In [ ]:
x = int(input())

sum = 0
i = 1

while sum + i <= x :
    sum += i
    i += 1 # 55가 됐을 때 1을 더한 후 stop되니 print(i-1) 해줄 것

print(i - 1)


55
10


In [ ]:
x = int(input())

sum = 0
i = 0

while sum < x :
    i += 1
    sum += i

print(i)

55
0 0
1 1
3 2
6 3
10 4
15 5
21 6
28 7
36 8
45 9
10


# 6080 : [기초-종합] 주사위 2개 던지기(설명)(py)


1부터 n까지, 1부터 m까지 숫자가 적힌 서로 다른 주사위 2개를 던졌을 때,
나올 수 있는 모든 경우를 출력해보자.

예시

```
for i in range(1, n+1) :
  for j in range(1, m+1) :
    print(i, j)
```

참고


위 코드는
바깥쪽의 i 값이 1부터 n까지 순서대로 바뀌는 각각의 동안에
안쪽의 j 값이 다시 1부터 m까지 변하며 출력되는 코드이다.

조건선택 실행구조 안에 다른 조건선택 실행구조를 넣어 처리할 수 있는 것과 마찬가지로
반복 실행구조 안에 다른 반복 실행구조를 넣어 처리할 수 있다.

원하는 형태로 실행 구조를 결합하거나 중첩시킬 수 있다.


In [ ]:
n, m = map(int, input().split())

for i in range(1, n+1):
    for j in range(1, m+1):
        print(i, j)

2 3
1 1
1 2
1 3
2 1
2 2
2 3


# 6081 : [기초-종합] 16진수 구구단 출력하기(py)

16진수(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E, F)를 배운
영일이는 16진수끼리 곱하는 16진수 구구단?에 대해서 궁금해졌다.

A, B, C, D, E, F 중 하나가 입력될 때,
1부터 F까지 곱한 16진수 구구단의 내용을 출력해보자.
(단, A ~ F 까지만 입력된다.)


```
 print('%X'%n, '*%X'%i, '=%X'%(n*i), sep='')
```
참고
print('%X'%n)    #n에 저장되어있는 값을 16진수(hexadecimal) 형태로 출력
작은 따옴표 2개를 사용해서 print(..., sep='') 으로 출력하면, 공백없이 모두 붙여 출력된다.
작은 따옴표 2개 '' 또는 큰 따옴표 2개 "" 는 아무 문자도 없는 빈문자열(empty string)을 의미한다.




## 16진수(hex), f-string 예시

In [ ]:
b = hex(145)
print(f'hex(145) = {b}')

hex(145) = 0x91


In [ ]:
print(f'{145:x}')

91


In [ ]:
print('%X' %145)

91


In [ ]:
pi = 3.141592653589793
print(f'{pi:.4f}') # 4째자리 까지 출력 = 5번째에서 반올림

3.1416


In [ ]:
import math
radius = 5
kk = "circle"
message = f"The area of a {kk} with radius {radius} is {math.pi * radius ** 2:.2f}."
print(message)

The area of a circle with radius 5 is 78.54.


In [ ]:
person = {"name": "Charlie", "age": 28}
message = f"{person['name']} is {person['age']} years old."
print(message)

Charlie is 28 years old.


In [ ]:
print(f'{15:X}')

F


In [ ]:
start = 1
end = int('F', 16) # F를 16진수로 변환
print(f'range({start}, {end + 1})')

range(1, 16)


In [ ]:
for i in range(start, end + 1):
    print(f'{i:X}', end = ' ')

1 2 3 4 5 6 7 8 9 A B C D E F 

In [ ]:
hex_val = 'B'
decimal_val = int(hex_val, 16)
print(decimal_val)

11


- 최종 제출

In [ ]:
hex_value = str(input()) # A~F 사이 값  입력
decimal_value = int(hex_value, 16) # 16진수로 해석해서 10진수 정수로 변환
start = 1
end = int('F', 16) # F를 16진수로 해석하여 10진수 정수로 변환

for i in range(start, end + 1):
    print(f'{decimal_value:X}*{i:X}={decimal_value*i:X}')

B
B*1=B
B*2=16
B*3=21
B*4=2C
B*5=37
B*6=42
B*7=4D
B*8=58
B*9=63
B*A=6E
B*B=79
B*C=84
B*D=8F
B*E=9A
B*F=A5


# 6082 : [기초-종합] 3 6 9 게임의 왕이 되자(설명)(py)

친구들과 함께 3 6 9 게임을 하던 영일이는 잦은 실수 때문에 계속해서 벌칙을 받게 되었다.

3 6 9 게임의 왕이 되기 위한 369 마스터 프로그램을 작성해 보자.

** 3 6 9 게임은?
여러 사람이 순서를 정한 후, 순서대로 수를 부르는 게임이다.

만약 3, 6, 9 가 들어간 수를 자신이 불러야 하는 상황이라면, 수를 부르는 대신 "박수(X)" 를 쳐야 한다.

33과 같이 3,6,9가 두 번 들어간 수 일때, "짝짝"과 같이 박수를 두 번 치는 형태도 있다.

참고
```
for i in range(1, n+1) :
  if i%10==3 :
    print("X", end=' ')    #출력 후 공백문자(빈칸, ' ')로 끝냄
```

입력


30 보다 작은 정수 1개가 입력된다.
(1 ~ 29)



Ex) 입력:9 -> 출력: 1 2 X 4 5 X 7 8 X

In [ ]:
sum(str(33).count(d) for d in '369')

2

## str.count 예시

In [ ]:
for i in range(1, 30):
    if sum(str(i).count(d) for d in '369') == 1 :
        print('X', end = ' ')
    elif sum(str(i).count(d) for d in '369') == 2:
        print('XX', end = ' ')
    else :
        print(i, end = ' ')


# 근데 이 코드는 자릿수 늘어날 때마다 elif == 3, 4, 5 ... 계속 추가해줘야함

1 2 X 4 5 X 7 8 X 10 11 12 X 14 15 X 17 18 X 20 21 22 X 24 25 X 27 28 X 

In [ ]:
for i in range(1, 34):
    temp = sum(str(i).count(d) for d in '369')
    if temp != 0 :
        print('X'*temp, end = ' ')
    else :
        print(i, end = ' ')

1 2 X 4 5 X 7 8 X 10 11 12 X 14 15 X 17 18 X 20 21 22 X 24 25 X 27 28 X X X X XX 

- 최종 제출

In [ ]:
end = int(input())

for i in range(1, end+1) :
    temp = sum(str(i).count(d) for d in '369')
    if temp != 0 :
        print('X'*temp, end = ' ')
    else :
        print(i, end = ' ')

9
1 2 X 4 5 X 7 8 X 

# 6083 : [기초-종합] 빛 섞어 색 만들기(설명)(py)

빨강(red), 초록(green), 파랑(blue) 빛을 섞어 여러 가지 다른 색 빛을 만들어 내려고 한다.

빨강(r), 초록(g), 파랑(b) 각 빛의 가짓수가 주어질 때,

주어진 rgb 빛들을 섞어 만들 수 있는 모든 경우의 조합(r g b)과 만들 수 있는 색의 가짓 수를 계산해보자.  

* 모니터, 스마트폰과 같은 디스플레이에서 각 픽셀의 색을 만들어내기 위해서 r, g, b 색을 조합할 수 있다.

* 픽셀(pixel)은 그림(picture)을 구성하는 셀(cell)에서 이름이 만들어졌다.

입력: 빨녹파(r, g, b) 각 빛의 가짓수가 공백을 두고 입력된다.

예를 들어, 3 3 3 은 빨녹파 빛에 대해서 각각 0~2까지 3가지 색이 있음을 의미한다.

0 <= r,g,b <= 127

출력: 만들 수 있는 rgb 색의 정보를 오름차순(계단을 올라가는 순, 12345... abcde..., 가나다라마...)으로
줄을 바꿔 모두 출력하고, 마지막에 그 개수를 출력한다.

## itertools 예시

In [ ]:
# 리스트 요소들 조합
import itertools

elements = ['A', 'B', 'C']
r = 2

combi = list(itertools.combinations(elements, r))
print(combi)
print()

for i in combi :
    print(i)

[('A', 'B'), ('A', 'C'), ('B', 'C')]

('A', 'B')
('A', 'C')
('B', 'C')


In [ ]:
# 문자열 조합
import itertools

elements = 'ABCD'
r = 3

combi = list(itertools.combinations(elements, r))
print(combi)

for i in combi:
    print(i, ''.join(i))

[('A', 'B', 'C'), ('A', 'B', 'D'), ('A', 'C', 'D'), ('B', 'C', 'D')]
('A', 'B', 'C') ABC
('A', 'B', 'D') ABD
('A', 'C', 'D') ACD
('B', 'C', 'D') BCD


In [ ]:
# 데카르트 곱
import itertools

tuple_r = (0, 1)
tuple_g = (0, 1)
tuple_b = (0, 1)

# itertools.product를 사용하여 모든 가능한 조합 생성
combinations = list(itertools.product(tuple_r, tuple_g, tuple_b))

print(combinations)

[(0, 0, 0), (0, 0, 1), (0, 1, 0), (0, 1, 1), (1, 0, 0), (1, 0, 1), (1, 1, 0), (1, 1, 1)]


## 리스트 컴프리헨션

In [ ]:
# 0부터 9까지의 수를 포함하는 리스트
a = [i for i in range(10)]
print(a)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]


In [ ]:
# 리스트 컴프리헨션 안 쓴 경우
a = []
for i in range(10) :
    a.append(i)
print(a)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]




---



In [ ]:
r, g, b = map(int, input().split(' '))

2 2 2


In [ ]:
list_r = [r for r in range(r)]
print(list_r)
list_g = [g for g in range(g)]
print(list_g)
list_b = [b for b in range(b)]
print(list_b)

[0, 1]
[0, 1]
[0, 1]


In [ ]:
import itertools

combi = list(itertools.product(list_r, list_g, list_b))
combi

[(0, 0, 0),
 (0, 0, 1),
 (0, 1, 0),
 (0, 1, 1),
 (1, 0, 0),
 (1, 0, 1),
 (1, 1, 0),
 (1, 1, 1)]

In [ ]:
sum = 0

for i in combi:
    sum += 1
    print(*i, sep = '  ')
print(sum)

0  0  0
0  0  1
0  1  0
0  1  1
1  0  0
1  0  1
1  1  0
1  1  1
8


In [ ]:
# 최종 제출
import itertools

r, g, b = map(int, input().split(' '))

list_r = [r for r in range(0, r)]
list_g = [g for g in range(0, g)]
list_b = [b for b in range(0, b)]

combinations = list(itertools.product(list_r, list_g, list_b))
sorted_combi = sorted(combinations)

sum = 0
for i in sorted_combi :
    sum += 1
    print(*i, sep = ' ')
print(sum)

2 2 2
0 0 0
0 0 1
0 1 0
0 1 1
1 0 0
1 0 1
1 1 0
1 1 1
8


# 6084 : [기초-종합] 소리 파일 저장용량 계산하기(py)

소리가 컴퓨터에 저장될 때에는 디지털 데이터화 되어 저장된다.

마이크를 통해 1초에 적게는 수십 번, 많게는 수만 번 소리의 강약을 체크하고,
한 번씩 체크할 때 마다 그 값을 정수값으로 바꾸어 저장하는 방식으로 소리를 파일로 저장할 수 있다.

값을 저장할 때에는 비트를 사용하는 정도에 따라 세세한 녹음 정도를 결정할 수 있고,
좌우(스테레오) 채널로 저장하면 2배… 5.1채널이면 6배의 저장공간이 필요하고,
녹음 시간이 길면 그 만큼 더 많은 저장공간이 필요하다.

1초 동안 마이크로 소리강약을 체크하는 횟수를 h
(헤르쯔, Hz 는 1초에 몇 번? 체크하는가를 의미한다.)

한 번 체크한 값을 저장할 때 사용하는 비트수를 b
(2비트를 사용하면 0 또는 1 두 가지, 16비트를 사용하면 65536가지..)

좌우 등 소리를 저장할 트랙 개수인 채널 개수를 c
(모노는 1개, 스테레오는 2개의 트랙으로 저장함을 의미한다.)

녹음할 시간(초) s가 주어질 때,

필요한 저장 용량을 계산하는 프로그램을 작성해보자.

실제로, 일반적인 CD 음질(44.1KHz, 16bit, 스테레오)로 1초 동안 저장하려면

44100 * 16 * 2 * 1 bit의 저장공간이 필요한데,

44100*16*2*1/8/1024/1024 로 계산하면 약 0.168 MB 정도가 필요하다.

이렇게 녹음하는 방식을 PCM(Pulse Code Modulation) 방법이라고 하는데,

압축하지 않은 순수한(raw) 소리 데이터 파일은 대표적으로 *.wav 가 있다.

**
      8 bit(비트)           = 1byte(바이트)       # 8bit=1Byte

1024 Byte(210 byte) = 1KB(킬로 바이트)  # 1024Byte=1KB

1024 KB(210 KB)      = 1MB(메가 바이트)

1024 MB(210 MB)     = 1GB(기가 바이트)

1024 GB(210 GB)      = 1TB(테라 바이트)

입력: h, b, c, s 가 공백을 두고 입력된다.

h는 48,000이하, b는 32이하(단, 8의배수), c는 5이하, s는 6,000이하의 자연수이다.

ex) 44100 16 2 10

출력: 필요한 저장 공간을 MB 단위로 바꾸어 출력한다.

단, 소수점 첫째 자리까지의 정확도로 출력하고 MB를 공백을 두고 출력한다.

ex) 1.7 MB

In [ ]:
h, b, c, s = map(int, input().split(' '))
print(f'{h * b * c * s / 8 / 1024 / 1024 :.1f} MB')

1.7 MB


# 6085 : [기초-종합] 그림 파일 저장용량 계산하기(py)
이미지가 컴퓨터에 저장될 때에도 디지털 데이터화 되어 저장된다.

가장 기본적인 방법으로는 그림을 구성하는 한 점(pixel, 픽셀)의 색상을

빨강(r), 초록(g), 파랑(b)의 3가지의 빛의 세기 값으로 따로 변환하여 저장하는 것인데,

예를 들어 r, g, b 각 색에 대해서 8비트(0~255, 256가지 가능)씩을 사용한다고 하면,

한 점의 색상은 3가지 r, g, b의 8비트+8비트+8비트로 총 24비트로 표현해서

총 2^24 가지의 서로 다른 빛의 색깔을 사용할 수 있는 것이다.

그렇게 저장하는 점을 모아 하나의 큰 이미지를 저장할 수 있게 되는데,

1024 * 768 사이즈에 각 점에 대해 24비트로 저장하면 그 이미지를 저장하기 위한
저장 용량을 계산할 수 있다.

이렇게 이미지의 원래(raw) 데이터를 압축하지 않고 그대로 저장하는 대표적인 이미지 파일이
*.bmp 파일이며, 비트로 그림을 구성한다고 하여 비트맵 방식 또는 래스터 방식이라고 한다.

이미지의 가로 해상도 w, 세로 해상도 h, 한 픽셀을 저장하기 위한 비트 b 가 주어질 때,

압축하지 않고 저장하기 위해 필요한 저장 용량을 계산하는 프로그램을 작성해 보자.

예를 들어

일반적인 1024 * 768 사이즈(해상도)의 각점에 대해

24비트(rgb 각각 8비트씩 3개)로 저장하려면

1024 * 768 * 24 bit의 저장공간이 필요한데,

1024*768*24/8/1024/1024 로 계산하면 약 2.25 MB 정도가 필요하다.

실제 그런지 확인하고 싶다면, 간단한 그림 편집/수정 프로그램을 통해 확인할 수 있다.

입력: w, h, b 가 공백을 두고 입력된다.

단, w, h는 모두 정수이고 1~1024 이다. b는 40이하의 4의 배수이다.

출력: 필요한 저장 공간을 MB 단위로 바꾸어 출력한다.

단, 소수점 셋째 자리에서 반올림하여 둘째 자리까지 출력한다.

In [ ]:
w, h, b = map(int, input().split(' '))
print(f'{w * h * b / 8 / 1024 / 1024 :.2f} MB')

1024 768 24
2.25 MB


# 6086 : [기초-종합] 거기까지! 이제 그만~(설명)(py)

1, 2, 3 ... 을 순서대로 계속 더해 합을 만드는데,
그 합이 입력한 정수보다 작을 동안만 계속 더하는 프로그램을 작성해보자.

즉, 1부터 n까지 정수를 하나씩 더해 합을 만드는데,
어디까지 더해야 입력한 수보다 같거나 커지는지 알아보고자 하는 문제이다.

하지만, 이번에는 그 때 까지의 합을 출력해야 한다.

예를 들어, 57을 입력하면

1+2+3+...+8+9+10=55에서 그 다음 수인 11을 더해 66이 될 때,

그 값 66이 출력되어야 한다.

예시


```
while True :
  s += c
  c += 1
  if s>=n :
    break

print(s)

```
참고

조건문이나 반복문의 코드블록 안에서 break 가 실행되면,
반복실행을 중단(break)하고, 가장 가까운 반복 블록의 밖으로 빠져나간다.


입력: 언제까지 합을 계산할 지, 정수 1개를 입력받는다.

단, 입력되는 자연수는 100,000,000이하이다.

출력: 1, 2, 3, 4, 5 ... 순서대로 계속 더해가다가, 그 합이 입력된 정수보다 커지거나 같아지는 경우,

그때까지의 합을 출력한다.

In [ ]:
sum = 0
n = 57
i = 1

while True:
    sum += i
    i += 1
    if sum >= n:
        break
print(sum, i)


66 12


In [ ]:
sum = 0
n = 57
i = 1
while sum < n:
    sum += i
    i += 1
print(sum, i)


66 12


In [ ]:
# 최종 제출

n = int(input())
sum = 0
i = 1

while sum < n :
    sum += i
    i += 1
print(sum)

57
66


# 6087 : [기초-종합] 3의 배수는 통과(설명)(py)

1부터 입력한 정수까지 1씩 증가시켜 출력하는 프로그램을 작성하되,
3의 배수인 경우는 출력하지 않도록 만들어보자.

예를 들면,
1 2 4 5 7 8 10 11 13 14 ...
와 같이 출력하는 것이다.

예시


```
for i in range(1, n+1) :
  if i%2==0 :
    continue            #다음 반복 단계로 넘어간다.
  print(i, end=' ')    #i가 짝수가 아닐 때만 실행된다.
```
위 코드는 홀 수만 출력하는 예시이다.

참고
조건문이나 반복문의 코드블록 안에서 continue 가 실행되면,
반복 블록 안에 있는 나머지 부분을 실행하지 않고, 다음 반복 단계로 넘어간다.
즉, 반복 블록의 나머지 부분은 실행되지 않고, 다음 단계의 반복을 계속(continue)하는 것이다.




In [ ]:
n = int(input())
for i in range(1, n+1) :
    if i%3 == 0 :
        continue
    print(i, end = ' ')

30
1 2 4 5 7 8 10 11 13 14 16 17 19 20 22 23 25 26 28 29 

In [ ]:
# 간소화

print(' '.join(str(i) for i in range(1, int(input()) + 1) if i % 3 != 0))

30
1 2 4 5 7 8 10 11 13 14 16 17 19 20 22 23 25 26 28 29


# 6088 : [기초-종합] 수 나열하기1(py)

어떤 규칙에 따라 수를 순서대로 나열한 것을 수열(sequences)이라고 한다.

예를 들어

1 4 7 10 13 16 19 22 25 ... 은

1부터 시작해 이전에 만든 수에 3을 더해 다음 수를 만든 수열이다.

이러한 것을 수학에서는 앞뒤 수들의 차이가 같다고 하여

등차(차이가 같다의 한문 말) 수열이라고 한다. (등차수열 : arithmetic progression/sequence)

수열을 알게 된 영일이는 갑자기 궁금해졌다.

"그럼.... 123번째 나오는 수는 뭘까?"

영일이는 프로그램을 만들어 더 큰 수도 자동으로 계산하고 싶어졌다.

시작 값(a), 등차(d), 몇 번째인지를 나타내는 정수(n)가 입력될 때
n번째 수를 출력하는 프로그램을 만들어보자.

In [ ]:
a, d, n = map(int, input().split())

print(a + (n-1)*d)

1 3 5
13


# 6089 : [기초-종합] 수 나열하기2(py)

어떤 규칙에 따라 수를 순서대로 나열한 것을 수열(sequences)이라고 한다.

예를 들어
2 6 18 54 162 486 ... 은
2부터 시작해 이전에 만든 수에 3을 곱해 다음 수를 만든 수열이다.

이러한 것을 수학에서는 앞뒤 수들의 비율이 같다고 하여
등비(비율이 같다의 한문 말) 수열이라고 한다. (등비수열 : geometric progression/sequence)

등비 수열을 알게된 영일이는 갑자기 궁금해졌다.
"그럼.... 13번째 나오는 수는 뭘까?"
영일이는 프로그램을 만들어 더 큰 수도 자동으로 계산하고 싶어졌다.

시작 값(a), 등비(r), 몇 번째인지를 나타내는 정수(n)가 입력될 때
n번째 수를 출력하는 프로그램을 만들어보자.


In [ ]:
a, r, n = map(int, input().split(' '))

print( a * (r**(n-1)) )

2 3 7


1458

# 6090 : [기초-종합] 수 나열하기3(py)

어떤 규칙에 따라 수를 순서대로 나열한 것을 수열(sequences)이라고 한다.

예를 들어
1 -1 3 -5 11 -21 43 ... 은
1부터 시작해 이전에 만든 수에 -2를 곱한 다음 1을 더해 다음 수를 만든 수열이다.

이런 이상한 수열을 알게 된 영일이는 또 궁금해졌다.
"그럼.... 13번째 나오는 수는 뭘까?"

영일이는 물론 수학을 아주 잘하지만 이런 문제는 본 적이 거의 없었다...
그래서 프로그램을 만들어 더 큰 수도 자동으로 계산하고 싶어졌다.

시작 값(a), 곱할 값(m), 더할 값(d), 몇 번째인지를 나타내는 정수(n)가 입력될 때,
n번째 수를 출력하는 프로그램을 만들어보자.

- 공식: n번째 항을 An, n-1번째 항을 An-1이라 하면
    
    An = m * An-1 + d

In [ ]:
a, m, d, n = map(int, input().split())
result = [a]

for i in range(1, n) :
    print('An:', a, 'i:', i)
    a = m * a + d
    result.append(a)
print('n번째 수:', a)
print('A1,...,An:', result)

1 -2 1 8
An: 1 i: 1
An: -1 i: 2
An: 3 i: 3
An: -5 i: 4
An: 11 i: 5
An: -21 i: 6
An: 43 i: 7
n번째 수: -85
A1,...,An: [1, -1, 3, -5, 11, -21, 43, -85]


In [ ]:
# 최종 제출
a, m, d, n = map(int, input().split())

for i in range(1, n) :
    a = m * a + d
print(a)

1 -2 1 8
-85


In [ ]:
1 -2 1 8

In [ ]:
a,m,d,n = map(int, input().split())
for i in range(n-1):
    a = a*m+d
print(a)

1 -2 1 8
-85


# 6091 : [기초-종합] 함께 문제 푸는 날(설명)(py)

온라인 채점시스템에는 초등학생, 중고등학생, 대학생, 대학원생,
일반인, 군인, 프로그래머, 탑코더 등 아주 많은 사람들이 들어와 문제를 풀고 있는데,

실시간 채점 정보는 메뉴의 채점기록(Judge Status)을 통해 살펴볼 수 있다.

자! 여기서...잠깐..
같은 날 동시에 가입한 3명의 사람들이 온라인 채점시스템에 들어와 문제를 푸는 날짜가
매우 규칙적이라고 할 때, 다시 모두 함께 문제를 풀게 되는 그날은 언제일까?

예를 들어 3명이 같은 날 가입/등업하고, 각각 3일마다, 7일마다, 9일마다
한 번씩 들어온다면, 처음 가입하고 63일 만에 다시 3명이 함께 문제를 풀게 된다.

갑자기 힌트?
왠지 어려워 보이지 않는가?
수학에서 배운 최소공배수를 생각한 사람들도 있을 것이다. 하지만, 정보에서 배우고 경험하는
정보과학의 세상은 때때로 컴퓨터의 힘을 빌려 간단한 방법으로 해결할 수 있게 한다.

아래의 코드를 읽고 이해한 후 도전해 보자.
day는 날 수, a/b/c는 방문 주기이다.



```
d = 1
while d%a!=0 or d%b!=0 or d%c!=0 :
  d += 1
print(d)
```
물론, 아주 많은 다양한 방법이 있을 수 있다.

정보과학의 문제해결에 있어서 정답은?
하나가 아니라 주어진 시간/기억공간으로 정확한 결과를 얻을 수 있는 모든 방법이다.

따라서, 모든 문제들에는 정답이 하나뿐만이 아니다.
새로운, 더 빠른, 더 간단한 방법을 다양하게 생각해보고 여러가지 방법으로 도전해 볼 수 있다.



In [ ]:
import math
a, b, c = map(int, input().split())
lcm = math.lcm(a, b, c)
print(lcm)

3 7 9
63


# 6092 : [기초-리스트] 이상한 출석 번호 부르기1(설명)(py)

정보 선생님은 수업을 시작하기 전에 이상한 출석을 부른다.

선생님은 출석부를 보고 번호를 부르는데,
학생들의 얼굴과 이름을 빨리 익히기 위해 번호를 무작위(랜덤)으로 부른다.

그리고 얼굴과 이름이 잘 기억되지 않는 학생들은 번호를 여러 번 불러
이름과 얼굴을 빨리 익히려고 하는 것이다.

출석 번호를 n번 무작위로 불렀을 때, 각 번호(1 ~ 23)가 불린 횟수를 각각 출력해보자.

In [ ]:
n = int(input()) # 출석 번호를 부른 횟수
num = list(map(int, input().split())) # 무작위로 부른 n개의 번호(1 ~ 23)가 공백을 두고 순서대로 입력
result = [0] * 23 #  count할 list 생성. 번호 (1~23까지 있음)

for i in range(n) :
    result[num[i]-1] += 1
print(result)

print(' '.join(str(result[i]) for i in range(len(result))))

10
1 3 2 2 5 6 7 4 5 9
[1, 2, 1, 1, 2, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
1 2 1 1 2 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0


In [ ]:
#최종 제출

n = int(input())
num = list(map(int, input().split()))
result = [0] * 23

for i in range(n) :
    result[num[i]-1] += 1

print(' '.join(str(result[i]) for i in range(len(result))))

10
1 3 2 2 5 6 7 4 5 9
1 2 1 1 2 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0


# 6093 : [기초-리스트] 이상한 출석 번호 부르기2(py)

정보 선생님은 수업을 시작하기 전에 이상한 출석을 부른다.

학생들의 얼굴과 이름을 빨리 익히기 위해 번호를 무작위(랜덤)으로 부르는데,
영일이는 선생님이 부른 번호들을 기억하고 있다가 거꾸로 불러보는 것을 해보고 싶어졌다.

출석 번호를 n번 무작위로 불렀을 때, 부른 번호를 거꾸로 출력해 보자.

예시



```
for i in range(n-1, -1, -1) :
  print(a[i], end=' ')
```

번호를 부른 순서를 리스트에 순서대로 기록해 두었다가, 기록한 값들을 거꾸로 출력하면 된다.

range(시작, 끝, 증감) #시작 수는 포함, 끝 수는 포함하지 않음. [시작, 끝)

range(n-1, -1, -1) #n-1, n-2, ..., 3, 2, 1, 0


In [ ]:
for i in range(10, 1, -1):
  print(i)

10
9
8
7
6
5
4
3
2


In [ ]:
n = int(input())
a = list(map(int,input().split()))

for i in range(n-1, -1, -1) : # a의 인덱스는 0~9니까 n-1, -1
    print(a[i], end = ' ')

10
10 4 2 3 6 6 7 9 8 5
5 8 9 7 6 6 3 2 4 10 

# 6094 : [기초-리스트] 이상한 출석 번호 부르기3(py)

정보 선생님은 오늘도 이상한 출석을 부른다.

영일이는 오늘도 다른 생각을 해보았다.
출석 번호를 다 부르지는 않은 것 같은데... 가장 빠른 번호가 뭐였지?

출석 번호를 n번 무작위로 불렀을 때, 가장 빠른 번호를 출력해 보자.

단,
첫 번째 번호와 마지막 번호가 몇 번인지는 아무도 모른다.
음수(-) 번호, 0번 번호도 있을 수 있다.

참고
리스트에 출석 번호를 기록해 두었다가, 그 중에서 가장 작은 값을 찾아내면 된다.
그런데, 가장 작은 값은 어떻게 어떤 것과 비교하고, 어떻게 찾아야 할까?

In [ ]:
n = int(input())
a = list(map(int, input().split()))

print(sorted(a)[0])

10
10 4 2 3 6 6 7 9 8 5
2


In [ ]:
a = [-1, 0, 4]
sorted(a)[0]

-1

# 6095 : [기초-리스트] 바둑판에 흰 돌 놓기(설명)(py)

기숙사 생활을 하는 학교에서 어떤 금요일(전원 귀가일)에는 모두 집으로 귀가를 한다.

오랜만에 집에 간 영일이는 아버지와 함께 두던 매우 큰 오목에 대해서 생각해 보다가
"바둑판에 돌을 올린 것을 프로그래밍 할 수 있을까?"하고 생각하였다.

바둑판(19 * 19)에 n개의 흰 돌을 놓는다고 할 때,
n개의 흰 돌이 놓인 위치를 출력하는 프로그램을 작성해보자.



```
d=[]                        #대괄호 [ ] 를 이용해 아무것도 없는 빈 리스트 만들기
for i in range(20) :
  d.append([])         #리스트 안에 다른 리스트 추가해 넣기
  for j in range(20) :
    d[i].append(0)    #리스트 안에 들어있는 리스트 안에 0 추가해 넣기

n = int(input())
for i in range(n) :
  x, y = input().split()
  d[int(x)][int(y)] = 1

for i in range(1, 20) :
  for j in range(1, 20) :
    print(d[i][j], end=' ')    #공백을 두고 한 줄로 출력
  print()                          #줄 바꿈
```

참고
리스트가 들어있는 리스트를 만들면?

가로번호, 세로번호를 사용해 2차원 형태의 데이터처럼 쉽게 기록하고 사용할 수 있다.

리스트이름[번호][번호] 형식으로 저장되어있는 값을 읽고 쓸 수 있고, 더 확장한 n차원의 리스트도 만들 수 있다.



```
d=[]
for i in range(20) :
  d.append([])
  for j in range(20) :  
    d[i].append(0)
```
위와 같이, 모두 0이 채워진 2차원 리스트를 만드는 코드를 아래와 같은 방법으로 짧게 만들 수도 있다.

... [0 for j in range(20)]  #20개의 0이 들어간 [0, 0, 0, ... , 0, 0, 0] 리스트

아래처럼 작성하면 위와 같은 리스트가 20개가 들어간 리스트를 한 번에 만들어 준다.

d = [[0 for j in range(20)] for i in range(20)]

이러한 리스트 생성 방식을 List Comprehensions 라고 한다.


In [ ]:
d = [[0 for i in range(19)] for i in range(19)]
print(len(d), len(d[0])) # row, col

19 19


In [ ]:
n = int(input())

for i in range(n):
    x, y = map(int, input().split())
    d[x-1][y-1] = 1 # 행렬은 (1,1)부터 시작하니까

5
1 1 
2 2
3 3
4 4 
5 5


In [ ]:
d

[[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,

In [ ]:
# 공백을 두고 한 줄로 출력

for i in range(19):
    for j in range(19):
        print(d[i][j], end = ' ')
    print()

0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 


In [ ]:
# 최종 코드
d = [[0 for i in range(19)] for j in range(19)]

n = int(input())

for i in range(n):
    x, y = map(int, input().split())
    d[x-1][y-1] = 1

for i in range(19):
    for j in range(19):
        print(d[i][j], end = ' ')
    print()

5
1 1
2 2
3 3
4 4
5 5
1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 


# 6096 : [기초-리스트] 바둑알 십자 뒤집기(py)

부모님을 기다리던 영일이는 검정/흰 색 바둑알을 바둑판에 꽉 채워 깔아 놓고 놀다가...

"십(+)자 뒤집기를 해볼까?"하고 생각했다.

십자 뒤집기는
그 위치에 있는 모든 가로줄 돌의 색을 반대(1->0, 0->1)로 바꾼 후,

다시 그 위치에 있는 모든 세로줄 돌의 색을 반대로 바꾸는 것이다.

어떤 위치를 골라 집자 뒤집기를 하면, 그 위치를 제외한 가로줄과 세로줄의 색이 모두 반대로 바뀐다.

바둑판(19 * 19)에 흰 돌(1) 또는 검정 돌(0)이 모두 꽉 채워져 놓여있을 때,
n개의 좌표를 입력받아 십(+)자 뒤집기한 결과를 출력하는 프로그램을 작성해보자.

예시


```
for i in range(n) :
  x,y=input().split()
  for j in range(1, 20) :
    if d[j][int(y)]==0 :
      d[j][int(y)]=1
    else :
      d[j][int(y)]=0

    if d[int(x)][j]==0 :
      d[int(x)][j]=1
    else :
      d[int(x)][j]=0
```
참고
리스트가 들어있는 리스트를 만들면?

가로번호, 세로번호를 사용해 2차원 형태의 데이터처럼 쉽게 기록하고 사용할 수 있다.

리스트이름[번호][번호] 형식으로 저장되어있는 값을 읽고 쓸 수 있다.



## 2차원 배열 입력받기

In [ ]:
# 2차원 배열 입력받기 1

import numpy as np
d = np.zeros((19, 19))
print(d)

[[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.

In [ ]:
for i in range(19) :
    d[i,] = list(map(int, input().split()))
print(d)

0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
[[0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 

In [ ]:
# 2차원 배열 입력받기 축약형

n = 3
temp = [list(map(int, input().split())) for _ in range(n)]

0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0


In [ ]:
d = [list(map(int, input().split())) for _ in range(19)]
n = int(input())

for _ in range(n) :
    x, y = map(int, input().split())
    x -= 1
    y -= 1
    for i in range(19) :
        d[i][y] = 1 if d[i][y] == 0 else 0
    for j in range(19) :
        d[x][j] = 1 if d[x][j] == 0 else 0

for i in range(19) :
    for j in range(19) :
        print(d[i][j], end = ' ')
    print()

0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
2
10 10
12 12
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 

# 6097 : [기초-리스트] 설탕과자 뽑기(py)

부모님과 함께 놀러간 영일이는
설탕과자(설탕을 녹여 물고기 등의 모양을 만든 것) 뽑기를 보게 되었다.

길이가 다른 몇 개의 막대를 바둑판과 같은 격자판에 놓는데,

막대에 있는 설탕과자 이름 아래에 있는 번호를 뽑으면 설탕과자를 가져가는 게임이었다.
(잉어, 붕어, 용 등 여러 가지가 적혀있다.)

격자판의 세로(h), 가로(w), 막대의 개수(n), 각 막대의 길이(l),
막대를 놓는 방향(d:가로는 0, 세로는 1)과
막대를 놓는 막대의 가장 왼쪽 또는 위쪽의 위치(x, y)가 주어질 때,

격자판을 채운 막대의 모양을 출력하는 프로그램을 만들어보자.

첫 줄에 격자판의 세로(h), 가로(w) 가 공백을 두고 입력되고,

두 번째 줄에 놓을 수 있는 막대의 개수(n)

세 번째 줄부터 각 막대의 길이(l), 방향(d), 좌표(x, y)가 입력된다.

1 <= w, h <= 100

1 <= n <= 10

d = 0 or 1

1 <= x <= 100-h

1 <= y <= 100-w

In [ ]:
h, w = map(int, input().split()) # 가로 세로
n = int(input()) # 막대 수
l, d, x, y = map(int, input().split()) # 길이, 방향, 좌표

In [ ]:
h, w = 5, 5
n = 2
5 5
3
2 0 1 1
3 1 2 3
4 1 2 5

In [ ]:
h, w = map(int, input().split()) # 가로 세로
n = int(input()) # 막대 수
matrix = [[0 for _ in range(w)] for _ in range(h)]
matrix

5 5
3


[[0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0]]

In [ ]:
for _ in range(n) :
    l, d, x, y = map(int, input().split())
    x -= 1 # (0,0)부터 시작이니까
    y -= 1
    for i in range(l) :
      if 0 <= x <= h and 0 <= y <= w :
          matrix[x][y] = 1
          if d == 0 :
              y += 1
          else :
              x += 1
      else : continue

for row in matrix:

for i in range(h) :
    for j in range(w) :
        print(matrix[i][j], end = ' ')
    print()

2 0 1 1
3 1 2 3
4 1 2 5
1 1 0 0 0 
0 0 1 0 1 
0 0 1 0 1 
0 0 1 0 1 
0 0 0 0 1 


In [14]:
# 최종 제출
h, w = map(int, input().split()) # 가로 세로
n = int(input()) # 막대 수
matrix = [[0 for _ in range(w)] for _ in range(h)]

for _ in range(n):
    l, d, x, y = map(int, input().split())
    x -= 1
    y -= 1
    for i in range(l):
        if 0 <= x < h and 0 <= y < w:  # 범위 체크
            matrix[x][y] = 1
            if d == 0:
                y += 1
            else:
                x += 1

for row in matrix :
    print(' '.join(map(str, row)))

5 5
3
2 0 1 1
3 1 2 3
4 1 2 5
1 1 0 0 0
0 0 1 0 1
0 0 1 0 1
0 0 1 0 1
0 0 0 0 1


# 6098 : [기초-리스트] 성실한 개미(py)

영일이는 생명과학에 관심이 생겨 왕개미를 연구하고 있었다.

왕개미를 유심히 살펴보던 중 특별히 성실해 보이는 개미가 있었는데,
그 개미는 개미굴에서 나와 먹이까지 가장 빠른 길로 이동하는 것이었다.

개미는 오른쪽으로 움직이다가 벽을 만나면 아래쪽으로 움직여 가장 빠른 길로 움직였다.
(오른쪽에 길이 나타나면 다시 오른쪽으로 움직인다.)

이에 호기심이 생긴 영일이는 그 개미를 미로 상자에 넣고 살펴보기 시작하였다.

미로 상자에 넣은 개미는 먹이를 찾았거나, 더 이상 움직일 수 없을 때까지
오른쪽 또는 아래쪽으로만 움직였다.

미로 상자의 구조가 0(갈 수 있는 곳), 1(벽 또는 장애물)로 주어지고,
먹이가 2로 주어질 때, 성실한 개미의 이동 경로를 예상해보자.

단, 맨 아래의 가장 오른쪽에 도착한 경우, 더 이상 움직일 수 없는 경우, 먹이를 찾은 경우에는
더이상 이동하지 않고 그 곳에 머무른다고 가정한다.

미로 상자의 테두리는 모두 벽으로 되어 있으며,
개미집은 반드시 (2, 2)에 존재하기 때문에 개미는 (2, 2)에서 출발한다.

In [1]:
d = [list(map(int, input().split())) for _ in range(10)]
print(d)

1 1 1 1 1 1 1 1 1 1
1 0 0 1 0 0 0 0 0 1
1 0 0 1 1 1 0 0 0 1
1 0 0 0 0 0 0 1 0 1
1 0 0 0 0 0 0 1 0 1
1 0 0 0 0 1 0 1 0 1
1 0 0 0 0 1 2 1 0 1
1 0 0 0 0 1 0 0 0 1
1 0 0 0 0 0 0 0 0 1
1 1 1 1 1 1 1 1 1 1
[[1, 1, 1, 1, 1, 1, 1, 1, 1, 1], [1, 0, 0, 1, 0, 0, 0, 0, 0, 1], [1, 0, 0, 1, 1, 1, 0, 0, 0, 1], [1, 0, 0, 0, 0, 0, 0, 1, 0, 1], [1, 0, 0, 0, 0, 0, 0, 1, 0, 1], [1, 0, 0, 0, 0, 1, 0, 1, 0, 1], [1, 0, 0, 0, 0, 1, 2, 1, 0, 1], [1, 0, 0, 0, 0, 1, 0, 0, 0, 1], [1, 0, 0, 0, 0, 0, 0, 0, 0, 1], [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]


In [10]:
d = [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1], [1, 0, 0, 1, 0, 0, 0, 0, 0, 1], [1, 0, 0, 1, 1, 1, 0, 0, 0, 1], [1, 0, 0, 0, 0, 0, 0, 1, 0, 1], [1, 0, 0, 0, 0, 0, 0, 1, 0, 1], [1, 0, 0, 0, 0, 1, 0, 1, 0, 1], [1, 0, 0, 0, 0, 1, 2, 1, 0, 1], [1, 0, 0, 0, 0, 1, 0, 0, 0, 1], [1, 0, 0, 0, 0, 0, 0, 0, 0, 1], [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]

In [11]:
d

[[1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
 [1, 0, 0, 1, 0, 0, 0, 0, 0, 1],
 [1, 0, 0, 1, 1, 1, 0, 0, 0, 1],
 [1, 0, 0, 0, 0, 0, 0, 1, 0, 1],
 [1, 0, 0, 0, 0, 0, 0, 1, 0, 1],
 [1, 0, 0, 0, 0, 1, 0, 1, 0, 1],
 [1, 0, 0, 0, 0, 1, 2, 1, 0, 1],
 [1, 0, 0, 0, 0, 1, 0, 0, 0, 1],
 [1, 0, 0, 0, 0, 0, 0, 0, 0, 1],
 [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]

In [12]:
x, y = 1, 1 # 좌표는 (0, 0)부터 시작이니까

while True :
    if d[x][y] == 0 :
        d[x][y] = 9
    elif d[x][y] == 2 : # 먹이 찾은 경우
        d[x][y] = 9
        break

    if d[x][y+1] != 1 :
        y += 1
    elif d[x+1][y] != 1 :
        x += 1

    if (y < 8 and d[x][y+1] == 1) and (x < 8 and d[x+1][y] == 1) : # 더이상 갈 곳 없는 경우
        break

    # 경계 조건
    if x >= 8 or y >= 8 :
        break
print(d)

print('='*99)

for row in d :
    print(' '.join(map(str, row)))

[[1, 1, 1, 1, 1, 1, 1, 1, 1, 1], [1, 9, 9, 1, 0, 0, 0, 0, 0, 1], [1, 0, 9, 1, 1, 1, 0, 0, 0, 1], [1, 0, 9, 9, 9, 9, 9, 1, 0, 1], [1, 0, 0, 0, 0, 0, 9, 1, 0, 1], [1, 0, 0, 0, 0, 1, 9, 1, 0, 1], [1, 0, 0, 0, 0, 1, 9, 1, 0, 1], [1, 0, 0, 0, 0, 1, 0, 0, 0, 1], [1, 0, 0, 0, 0, 0, 0, 0, 0, 1], [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]
1 1 1 1 1 1 1 1 1 1
1 9 9 1 0 0 0 0 0 1
1 0 9 1 1 1 0 0 0 1
1 0 9 9 9 9 9 1 0 1
1 0 0 0 0 0 9 1 0 1
1 0 0 0 0 1 9 1 0 1
1 0 0 0 0 1 9 1 0 1
1 0 0 0 0 1 0 0 0 1
1 0 0 0 0 0 0 0 0 1
1 1 1 1 1 1 1 1 1 1


In [ ]:
# 최종 코드

d = [list(map(int, input().split())) for _ in range(10)]

x, y = 1, 1

while True :
    if d[x][y] == 0 :
        d[x][y] = 9
    elif d[x][y] == 2 : # 먹이 찾은 경우
        d[x][y] = 9
        break

    if (d[x][y+1] == 1 and d[x+1][y] == 1) or (x > 8 or y > 8) :
        break

    if d[x][y+1] != 1 :
        y += 1
    elif d[x+1][y] != 1 :
        x += 1

for row in d :
    print(' '.join(map(str, row)))